server/src/main/java/org/elasticsearch/common/logging/LogConfigurator.java

```diff
@@ -269,8 +269,9 @@ public FileVisitResult visitFile(final Path file, final BasicFileAttributes attr
                     LogManager.getLogger("stderr"),
                     Level.WARN,
                     // MMapDirectory messages come from Lucene, suggesting to users as a warning that they should enable preview features in
-                    // the JDK
-                    List.of("MMapDirectory")
+                    // the JDK. Vector logs come from Lucene too, but only if the used explicitly disables the Vector API - no point warning
+                    // in this case.
+                    List.of("MMapDirectory", "VectorUtilProvider", "WARNING: Java vector incubator module is not readable")
                 ),
```

Comment:
This suppresses the two line warning from Lucene, if one was to disable the Vector API (by editing jvm.options).

```
[2023-06-01T15:15:29,430][WARN ][stderr                   ] [yamlRestTest-0] Jun 01, 2023 3:15:29 PM org.apache.lucene.util.VectorUtilProvider lookup
[2023-06-01T15:15:29,434][WARN ][stderr                   ] [yamlRestTest-0] WARNING: Java vector incubator module is not readable. For optimal vector performance, pass '--add-modules jdk.incubator.vector' to enable Vector API.
```